core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -919,16 +978,65 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
-                RelationshipLevel.Unforgivable -> addModifier(DiplomaticModifiers.DeclaredFriendshipWithOurEnemies, -15f)
-                RelationshipLevel.Enemy -> addModifier(DiplomaticModifiers.DeclaredFriendshipWithOurEnemies, -5f)
-                RelationshipLevel.Friend -> addModifier(DiplomaticModifiers.DeclaredFriendshipWithOurAllies, 5f)
-                RelationshipLevel.Ally -> addModifier(DiplomaticModifiers.DeclaredFriendshipWithOurAllies, 15f)
-                else -> {}
-            }
+            val relationshipLevel =  otherCiv().getDiplomacyManager(thirdCiv).relationshipIgnoreAfraid()
+            addModifier(when (relationshipLevel) {
+                RelationshipLevel.Unforgivable, RelationshipLevel.Enemy -> DiplomaticModifiers.DeclaredFriendshipWithOurEnemies
+                RelationshipLevel.Friend, RelationshipLevel.Ally -> DiplomaticModifiers.DeclaredFriendshipWithOurAllies
+                else -> DiplomaticModifiers.SignedDefensivePactWithOurAllies }, when (relationshipLevel) {
+                RelationshipLevel.Unforgivable -> -15f
+                RelationshipLevel.Enemy -> -5f
+                RelationshipLevel.Friend -> 5f
+                RelationshipLevel.Ally -> 15f
+                else -> {0f}
+            })
+
         }
     }
 
+    fun signDefensivePact(duration: Int) {
+        //Note: These modifiers are additive to the friendship modifiers
+        setModifier(DiplomaticModifiers.DefensivePact, 10f)
+        otherCivDiplomacy().setModifier(DiplomaticModifiers.DefensivePact, 10f)
+        setFlag(DiplomacyFlags.DefensivePact, duration)
+        otherCivDiplomacy().setFlag(DiplomacyFlags.DefensivePact, duration)
+        diplomaticStatus = DiplomaticStatus.DefensivePact;
+        otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.DefensivePact;
+
+        for (thirdCiv in getCommonKnownCivs().filter { it.isMajorCiv() }) {
+            thirdCiv.addNotification("[${civInfo.civName}] and [$otherCivName] have signed the Defensive Pact!",
+                NotificationCategory.Diplomacy, civInfo.civName, NotificationIcon.Diplomacy, otherCivName)
+            thirdCiv.getDiplomacyManager(civInfo).setDefensivePactBasedModifier()
+        }
+
+        // Ignore contitionals as triggerCivwideUnique will check again, and that would break
+        // UniqueType.ConditionalChance - 25% declared chance would work as 6% actual chance
+        for (unique in civInfo.getTriggeredUniques(UniqueType.TriggerUponSigningDefensivePact, StateForConditionals.IgnoreConditionals))
+            UniqueTriggerActivation.triggerCivwideUnique(unique, civInfo)
+        for (unique in otherCiv().getTriggeredUniques(UniqueType.TriggerUponSigningDefensivePact, StateForConditionals.IgnoreConditionals))
+            UniqueTriggerActivation.triggerCivwideUnique(unique, otherCiv())
+    }
+
+    private fun setDefensivePactBasedModifier() {
+        removeModifier(DiplomaticModifiers.SignedDefensivePactWithOurAllies)
```

Comment:
Why are we removing these modifiers? If I had a defensive pact with your enemy, then with your sky, why should only the last one count?

Reply:
I kept the defensive pact behavior consistent with the declaration of friendship behavior. This is the same logic as in setFriendshipBasedModifier().